### PR TITLE
Votebox implementation

### PR DESCRIPTION
**Pull request by ansxor**
OK being highlighted
BAD being highlighted

The votebox should work properly, however there is no live updating yet. If you can inform me on how to use the listener with votes (and also in your system overall lol), I can try and implement it myself but you can also do it yourself if you think it will be easier. The counts for the votes would be under $voteCount_g, $voteCount_o and $voteCount_b.

I can also revert the styling so that it just uses a lime background instead of the gradients that I chose for no reason. I can also revert the names of the votes to BAD, OK and GREAT if they are not clear enough.

### CODE REVIEW

**Comment by randomouscrap98** [comment]
Oh votes... I don't think we can listen for updates to those yet

File: draw.js [hunk 1]
```diff
@@ -794,6 +795,92 @@ sidebarComment: function(comment) {
...
+		if (oldButton &&
+				oldButton.hasAttribute('data-selected') &&
+				button.hasAttribute('data-selected'))
+			vote=undefined
+		Req.setVote(page.id, vote, function(e, resp) {
+			// in case the vote fails when user is blocked from voting
+			if (resp) {
+				// if the vote was already toggled, then remove highlight
+				var replaceVote = function(q, x) {
+					var c = document.querySelector(q);
```
Comment (12Me21) [inline draw.js:823]: trailing whitespace

File: draw.js [hunk 2]
```diff
@@ -794,6 +795,92 @@ sidebarComment: function(comment) {
...
+					button.setAttribute('data-selected', 'true')
+					replaceVote('#\\$voteCount_' + state, 1)
+				}
+			}
+		})
+		e.stopPropagation()
+	}
+
+	var div = document.createElement('div')
+	div.className += 'buttonContainer rightAlign item loggedIn'
```
Comment (12Me21) [inline draw.js:847]: use a leading space in the string when appending to classname

File: draw.js [hunk 3]
```diff
@@ -794,6 +795,92 @@ sidebarComment: function(comment) {
...
+	
+	var b = document.createElement('button')
+	b.id = "$voteButton_" + state
+	b.className += 'voteButton'
```
Comment (12Me21) [inline draw.js:851]: classname

File: draw.js [hunk 4]
```diff
@@ -794,6 +795,97 @@ sidebarComment: function(comment) {
...
+// which is clearly an awful way of doing things. I don't
+// understand how listener is implemented 	in this, so I don't
+// feel like touching it.
+voteButton: function(disptext, state, page) {
+	var div = document.createElement('div')
+	div.className += ' buttonContainer rightAlign item'
+	
+	var b = document.createElement('button')
+	b.className += ' voteButton'
+	console.log(state, page.about)
```
Comment (12Me21) [inline draw.js:811]: console.log

File: draw.js [hunk 5]
```diff
@@ -794,6 +795,97 @@ sidebarComment: function(comment) {
...
+	label.textContent = disptext
+	b.appendChild(label)
+	
+	var count = document.createElement('div')
+	count.className = ' voteCount'
+	count.setAttribute('data-vote', state)
+	count.textContent = page.about.votes[state].count
+	b.appendChild(count)
+
+	console.log(div);
```
Comment (12Me21) [inline draw.js:826]: console.log

File: draw.js [hunk 6]
```diff
@@ -794,6 +795,93 @@ sidebarComment: function(comment) {
...
+	count.setAttribute('data-vote', state)
+	count.textContent = page.about.votes[state].count
+	b[1].appendChild(count)
+	
+	return b[0]
+},
+
+voteBox: function (page) {
+	var element = document.createElement('div')
+	element.className += ' buttonContainer rightAlign'
```
Comment (12Me21) [inline draw.js:827]: class should be "item rightAlign", not buttoncontainer

File: draw.js [hunk 7]
```diff
@@ -794,6 +795,93 @@ sidebarComment: function(comment) {
...
 	
+},
+
+// FIXME: There is no live updating of the vote count. It only adds
+// the user's count to the user vote when the user votes,
+// which is clearly an awful way of doing things. I don't
+// understand how listener is implemented 	in this, so I don't
+// feel like touching it.
+voteButton: function(disptext, state, page) {
+	var b = button()
```
Comment (12Me21) [inline draw.js:806]: b[0] (button container) needs to have class 'item' too

**Pull request merged**